src/timebar/components/date-selector.js

```diff
@@ -1,32 +1,97 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types'
+import cx from 'classnames'
 import styles from './date-selector.module.css'
 import { ReactComponent as IconArrowUp } from '../icons/arrowUp.svg'
 import { ReactComponent as IconArrowDown } from '../icons/arrowDown.svg'
 
 class DateSelector extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      value: props.value,
+      error: false,
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
It is deprecated and not a good practice, but was the easiest way to reuse the existing code.